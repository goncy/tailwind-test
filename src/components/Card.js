import React from "react";

const Card = ({ image, tags, title, children, ...props }) => (
  <div className="max-w-sm rounded overflow-hidden shadow-lg" {...props}>
    {image && <img alt={title} className="w-full h-64" src={image} />}
    <div className="px-6 py-4">
      <div className="font-bold text-xl mb-2">{title}</div>
      <p className="text-gray-700 text-base">{children}</p>
    </div>
    {tags && Boolean(tags.length) && (
      <div className="px-6 py-4">
        {tags.map(tag => (
          <span
            key={tag}
            className="inline-block bg-gray-200 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
          >
            #{tag}
          </span>
        ))}
      </div>
    )}
  </div>
);

export default Card;
