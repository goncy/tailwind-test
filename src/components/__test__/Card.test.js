import React from "react";
import { render, cleanup } from "@testing-library/react";

import Card from "../Card";

afterEach(cleanup);

it("shows the title", () => {
  const { baseElement } = render(
    <Card image="//placehold.it/128" tags={["sunset", "beach"]} title="Live in Bali">
      If you really like beaches you can come here
    </Card>
  );

  expect(baseElement).toHaveTextContent("Live in Bali");
});

it("renders", () => {
  const { asFragment } = render(
    <Card image="//placehold.it/128" tags={["sunset", "beach"]} title="Live in Bali">
      If you really like beaches you can come here
    </Card>
  );

  expect(asFragment()).toMatchSnapshot();
});
