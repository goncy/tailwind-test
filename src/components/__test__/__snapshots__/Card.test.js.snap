// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<DocumentFragment>
  <div
    class="max-w-sm rounded overflow-hidden shadow-lg"
  >
    <img
      alt="Live in Bali"
      class="w-full h-64"
      src="//placehold.it/128"
    />
    <div
      class="px-6 py-4"
    >
      <div
        class="flex justify-between items-center mb-2"
      >
        <div
          class="font-bold text-xl"
        >
          Live in Bali
        </div>
        <div />
      </div>
      <p
        class="text-gray-700 text-base"
      >
        If you really like beaches you can come here
      </p>
    </div>
    <div
      class="px-6 py-4"
    >
      <span
        class="inline-block bg-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
      >
        #sunset
      </span>
      <span
        class="inline-block bg-gray-300 rounded-full px-3 py-1 text-sm font-semibold text-gray-700 mr-2"
      >
        #beach
      </span>
    </div>
  </div>
</DocumentFragment>
`;
